### PR TITLE
Cart page and functionality

### PR DESCRIPTION
**Pull request by sushil-mod**
# Added Cart Page and its Functionality 
how to check :- click on "login as a guest" then you redirect to home page go to "shop now" click on "add to cart" and then go to cart 
## what to check 

-  CartCard  :- src/components/card/CartCard.jsx
-  PriceDetailCard :- src/components/card/PriceDetailCard.jsx
-  Cart :- src/pages/cart/Cart.jsx

### Test 
email: adarshbalika@gmail.com
paswword: adarshbalika

### CODE REVIEW

File: src/components/card/CartCard.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import { useAthorizer } from '../../context/AuthorizerContext';
+
+function CartCard({product}) {
+  
+    const {_id , name , rating , orignalPrice , discount , imageUrl , qty} = product;
+    const { authState , authDispatch } = useAthorizer();
+
+    const productQtyHandler = async( productId , type ) => {
+        try {
+         const response = await axios.post(`/api/user/cart/${productId}`,{action:{type:`${type}`}},{headers:{authorization :authState.encodedToken }})
```
Comment (JantuDeb) [inline src/components/card/CartCard.jsx:12]: const {data} = await ....

**Review by JantuDeb — APPROVED** [review]

File: src/components/card/CartCard.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            console.log(error);
+        }
+    }
+
+    const removeFromCart = async(productId) =>{
+        try {
+                const response = await axios.delete(`/api/user/cart/${productId}`,{headers:{authorization :authState.encodedToken }})
+                authDispatch({type:"REMOVE_FROM_CART" , payload:response.data.cart});
+        } catch (error) {
+            console.log(error);
```
Comment (tusharkandpal) [inline src/components/card/CartCard.jsx:24]: Suggestion: Can use `console.error(error)` for error

File: src/components/card/ProductCard.jsx [hunk 3]
```diff
@@ -50,14 +57,14 @@ return <div key={_id}>
                         ({ discount }% OFF)
                     </h4>
                 </span>
-
             </div>
-            <button className="btn btn-shop card-fix-btn card-btn-pri ">
-                <FaShoppingCart /> Add to Cart </button>
-
+            { (authState.loginStatus && authState.cart.some((item)=>item._id===_id)) ? (<button className="btn btn-shop card-fix-btn card-btn-sec" onClick={()=>navigator("/cart") }  >
```
Comment (tusharkandpal) [inline src/components/card/ProductCard.jsx:61]: Suggestion: You can destrucure `authState` above

File: src/components/navbar/Navbar.jsx [hunk 4]
```diff
@@ -19,20 +19,31 @@ function Navbar (){
...
+                    </span> :
+                    <Link className="nav-login-link" to="/login">
+                        <button className="nav-login-btn">Login</button>
+                    </Link>
+            }
                         
             <ul className="nav-list">
                 <li className="nav-list-item">
-                    <Link className="nav-list-link icon-badge pos-rel" to="./screens/html/wishlist.html"><i
+                    <div className="nav-list-link icon-badge pos-rel"  style={{ cursor:"pointer" }} ><i
```
Comment (tusharkandpal) [inline src/components/navbar/Navbar.jsx:33]: Avoid inline-style instead can make a class to reuse

File: src/components/navbar/Navbar.jsx [hunk 5]
```diff
@@ -19,20 +19,31 @@ function Navbar (){
...
                         <div className="badge-number flex-center header-badge-color badge-on-icon pos-abt">3</div>
-                    </Link>
+                    </div>
                 </li>
                 <li className="nav-list-item">
-                    <Link className="nav-list-link icon-badge pos-rel" to="./screens/html/cart.html"><i
-                            className="fal fa-shopping-cart nav-link-icon"></i>
-                        <div className="badge-number flex-center header-badge-color badge-on-icon pos-abt">3</div>
-                    </Link>
+                    <div className="nav-list-link icon-badge pos-rel"  style={{ cursor:"pointer" }} >
```
Comment (tusharkandpal) [inline src/components/navbar/Navbar.jsx:39]: Here too

File: src/pages/auth/Login.jsx [hunk 6]
```diff
@@ -77,16 +59,17 @@ return <>
             </div>
 
             <div className="form-btn ">
-                <button className="btn login padd-sm" type="submit">Login</button>
+                <button className="btn login padd-sm margin-btm" type="submit">Login</button>
+            </div>
+            <div className="form-btn ">
+                <button className="nav-login-btn" onClick={(e)=>{ e.preventDefault(); loginSubmitHandler(e,guestUser) } }>Login as guest</button>
```
Comment (tusharkandpal) [inline src/pages/auth/Login.jsx:65]: You can remove `e.preventDefault()` here since you're re-invoking it inside `loginSubmitHandler` method

**Review by tusharkandpal — APPROVED** [review]
Good that you're using same context for cart & auth.

File: src/components/card/CartCard.jsx (hunk 1 above)
Comment (sushil-mod) [reply src/components/card/CartCard.jsx:12]: will do that  , thank you

File: src/components/card/CartCard.jsx (hunk 2 above)
Comment (sushil-mod) [reply src/components/card/CartCard.jsx:24]: yes ,i will change it

File: src/components/card/ProductCard.jsx (hunk 3 above)
Comment (sushil-mod) [reply src/components/card/ProductCard.jsx:61]: yeah , it will be easier

File: src/components/navbar/Navbar.jsx (hunk 4 above)
Comment (sushil-mod) [reply src/components/navbar/Navbar.jsx:33]: thanks for pointing out i will correct it

File: src/pages/auth/Login.jsx (hunk 6 above)
Comment (sushil-mod) [reply src/pages/auth/Login.jsx:65]: ooh forgot , thank you

**Pull request merged**